src/Integration.Vsix/CFamily/CLangAnalyzer.cs

```diff
@@ -159,18 +158,17 @@ private void RunAnalysis(IRequest request, IIssueConsumer consumer, IAnalysisSta
         {
             var analysisStartTime = DateTime.Now;
             statusNotifier?.AnalysisStarted(request.Context.File);
-            int issueCount = 0;
 
-            var handleMessage = consumer == null
-                ? (Action<Message>) (message => { })
-                : message => HandleMessage(message, request, consumer, ref issueCount);
+            var messageHandler = consumer == null
```

Comment:
As discussed offline, there needs to be a new `MessageHandler` instance per `IRequest` since it contains request-specific settings. I haven't created a test factory interface for this class yet; might do that as part of later refactoring of `CLangAnalyzer` once I've added the new functionality required for the current sprint.